saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+                Name of burnable material.
+            ``value``
+                `Materialflow` object holding composition and properties.
+
+        """
+
+    def run_depcode(self, cores, nodes):
+        """Runs OpenMC depletion simulation as a subprocess with the given
+        parameters.
+
+        Parameters
+        ----------
+        cores : int
+            Number of cores to use for depletion code run.
+        nodes : int
+            Number of nodes to use for depletion code run.
+        """
+        # need to add flow control for plots option
+        args = (
+            'mpiexec',
+            '-n',
+            str(nodes),
+            'python',
+            './deplete_openmc.py'
+            '-mat',
+            self.iter_matfile,
+            '-geo',
+            self.iter_inputfile['geometry'],
+            '-set',
+            self.iter_inputfile['settings'],
+            '-tal',
+            self.iter_inputfile['tallies'],
+            '-dep',
+            self.iter_inputfile['depletion_settings'])
+
+        print('Running %s' % (self.codename))
+        # Need to figure out how to adapt this to openmc
+        try:
+            subprocess.check_output(
+                args,
+                cwd=os.path.split(self.template_inputfile_path)[0],
+                stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as error:
+            print(error.output.decode("utf-8"))
+            raise RuntimeError('\n %s RUN FAILED\n see error message above'
+                               % (self.codename))
+        print('Finished OpenMC Run')
+
+    def switch_to_next_geometry(self):
+        """Switches the geometry file for the OpenMC depletion simulation to
+        the next geometry file in `geo_files`.
+        """
+        mats = openmc.Materials.from_xml(self.iter_matfile)
+        next_geometry = openmc.Geometry.from_xml(
+            path=self.geo_files.pop(0),
+            materials=mats)
+        next_geometry.export_to_xml(path=self.iter_inputfile['geometry'])
+        del mats, next_geometry
+
+    def write_depcode_input(self, reactor, dep_step, restart):
+        """ Writes prepared data into OpenMC input file(s).
+
+        Parameters
+        ----------
+        reactor : Reactor
+            Contains information about power load curve and cumulative
+            depletion time for the integration test.
+        dep_step : int
+            Current depletion time step.
+        restart : bool
+            Is the current simulation restarted?
+        """
+
+        if dep_step == 0 and not restart:
+            materials = openmc.Materials.from_xml(
+                self.template_inputfiles_path['materials'])
+            geometry = openmc.Geometry.from_xml(
+                self.template_inputfiles_path['geometry'], materials=materials)
+            settings = openmc.Settings.from_xml(
+                self.template_inputfiles_path['settings'])
+            settings.particles = self.npop
```

Comment:
The npop variable doesn't really sit well with me. If npop = particles, could you rename npop to particles?

Reply:
`npop` is serpent-like syntax. It existed in the code before this PR. Consider opening an issue about this if you feel strongly about this change. It should have it's own PR though since we are already changing many things in this one.